source/drivers/MicroBitCompassCalibrator.cpp

```diff
@@ -67,25 +242,27 @@ void MicroBitCompassCalibrator::calibrate(MicroBitEvent)
         uint8_t on;
     };
 
-    const int PERIMETER_POINTS = 12;
+    const int PERIMETER_POINTS = 25;
+
     const int PIXEL1_THRESHOLD = 200;
-    const int PIXEL2_THRESHOLD = 800;
+    const int PIXEL2_THRESHOLD = 680;
 
     wait_ms(100);
 
-	Matrix4 X(PERIMETER_POINTS, 4);
-    Point perimeter[PERIMETER_POINTS] = {{1,0,0}, {2,0,0}, {3,0,0}, {4,1,0}, {4,2,0}, {4,3,0}, {3,4,0}, {2,4,0}, {1,4,0}, {0,3,0}, {0,2,0}, {0,1,0}};
+    Point perimeter[PERIMETER_POINTS] = {{0,0,0}, {1,0,0}, {2,0,0}, {3,0,0}, {4,0,0}, {0,1,0}, {1,1,0}, {2,1,0}, {3,1,0}, {4,1,0}, {0,2,0}, {1,2,0}, {2,2,0}, {3,2,0}, {4,2,0}, {0,3,0}, {1,3,0}, {2,3,0}, {3,3,0}, {4,3,0}, {0,4,0}, {1,4,0}, {2,4,0}, {3,4,0}, {4,4,0}};
     Point cursor = {2,2,0};
 
     MicroBitImage img(5,5);
     MicroBitImage smiley("0,255,0,255,0\n0,255,0,255,0\n0,0,0,0,0\n255,0,0,0,255\n0,255,255,255,0\n");
+
+    CompassSample data[PERIMETER_POINTS];
     int samples = 0;
 
     // Firstly, we need to take over the display. Ensure all active animations are paused.
     display.stopAnimation();
-    display.scrollAsync("DRAW A CIRCLE");
+    display.scrollAsync("TILT TO FILL SCREEN");
```

Comment:
This new message is longer than before and so takes more time to scroll.  It can be very annoying especially when you've read it before!  I suggest 1) make it shorter, like "TILT TO FILL"; 2) allow the user to press button A or B to stop the message and go straight into the calibration game.

Reply:
Agreed. The slowness of calibration is the number 1 reason to avoid this feature. I suggest a distincitve little animation instead of text. Maybe some kind of radar-like anim.

Reply:
Text also doesn't work well globally with lots of language variants. Icons and animations are easier to communicate without strong language skills via a simple video or animation on our website.

Reply:
How about an arrow that spins around the screen to signify a compass, followed by a question mark? Play the animation once, then the first movement the device sees after that movement triggers the actual calibration.

Something like this...
```
let moved = false
input.onGesture(Gesture.Shake, () => {
    moved = true
})
basic.forever(() => {
    moved = false
    while (!(moved)) {
        images.arrowImage(ArrowNames.North).showImage(0)
        images.arrowImage(ArrowNames.NorthEast).showImage(0)
        images.arrowImage(ArrowNames.East).showImage(0)
        images.arrowImage(ArrowNames.SouthEast).showImage(0)
        images.arrowImage(ArrowNames.South).showImage(0)
        images.arrowImage(ArrowNames.SouthWest).showImage(0)
        images.arrowImage(ArrowNames.West).showImage(0)
        images.arrowImage(ArrowNames.NorthWest).showImage(0)
        basic.showString("?")
        basic.pause(250)
        basic.clearScreen()
        basic.pause(250)
        basic.showString("?")
        basic.pause(250)
        basic.clearScreen()
        basic.pause(250)
    }
    // Put actual compass cal routine here
    basic.showLeds(`
        . . # . .
        . . # . .
        . . # . .
        . . . . .
        . . . . .
        `)
    basic.pause(10000)
})
```

Reply:
I think in the first instance allowing button A+B to skip the message is best, because we have a real issue communicating to people what's happening with the compass calibration. @jamesadevine suggested this yesterday too. Just having a single button makes chances of users 'missing' the message too high. All of us approach this knowing exactly what's going on/about to happen but lots of people will be seeing it for the first time. 

We did toy with "Tilt to fill" which I think is probably fine too. Anyone object?

Reply:
@jaustin I'm not sure the text demystifies the process anymore than an animation, it will still be confusing for first time users; I like the idea of an animation more.

Reply:
@jaustin @DavidWhaleIET Got any new users to test an animation vs. text?

Reply:
@jamesadevine  So, in the testing we did with kids of text only "Tilt to fill screen" didn't need explanation in the same way "Draw a circle" did because they *way* do draw a circle wasn't clear, but the 'tilt' part explains how to fill the screen here - so that means the text alone worked, without video or other docs. 

The translation issue of the text is nasty though :/ It's a good reason to prefer icons over text long run. As far as I know we don't have any localisable text inside the pre-built components of micro:bit at the moment, and I think that's a 'good thing' in many ways. Does anyone know of previous discussions around localisation of the base components (IE MicroPython and DAL)?